### PR TITLE
High contrast defaults + color switching

### PR DESCRIPTION
**Pull request by DanBradbury**
@Petesta: The hope with this is to remove colors as they are currently done and replace it with a set of high contrast defaults that can be swapped on demand.

The original desire was to be able to switch each individual selection.. This would require a refactor to include a set of highlights (`poi`,`poi1`,etc) that hold the defaults plus a new storage system for highlights (groups according to highlighting colors probably makes the most sense)..

Anyways this is a WIP so I welcome any feedback

### CODE REVIEW

**Comment by DanBradbury** [comment]
@Petesta: should be ready for a final review (as a + I was able to get a `matchadd` branch to work)..

File: plugin/poi.vim [hunk 1]
```diff
@@ -1,32 +1,56 @@
...
-let s:match_base = ':match poi '
-au! VimEnter * execute ":autocmd InsertLeave * call <SID>MakeMatch()"
-au! ColorScheme * execute 'highlight poi ctermbg='.s:c_bg.' ctermfg='.s:c_fg.' guibg='s:g_bg.' guifg='.s:g_fg
-au! BufEnter * execute 'highlight poi ctermbg='.s:c_bg.' ctermfg='.s:c_fg.' guibg='s:g_bg.' guifg='.s:g_fg
-au! BufWinEnter * execute 'highlight poi ctermbg='.s:c_bg.' ctermfg='.s:c_fg.' guibg='s:g_bg.' guifg='.s:g_fg
-au! BufEnter * call <SID>MakeBuff()
-au! CursorHold * call <SID>MakeMatch()
-au! CursorMoved * call <SID>MakeMatch()
-au! CursorMovedI * call <SID>MakeMatch()
+let s:match_base = ':match poi1 '
```
Comment (Petesta) [inline plugin/poi.vim]: It appears you were aligning everything in your commits. For example look at lines in the "Highlight Lines" section of the README. Probably want to do the same here.

File: plugin/poi.vim [hunk 2]
```diff
@@ -1,32 +1,56 @@
...
+au! BufWinEnter * call <SID>ExecuteHighlight()
+au! CursorHold * call <SID>CleanupCrew()
+au! CursorMoved * call <SID>CleanupCrew()
+au! CursorMovedI * call <SID>CleanupCrew()
 
+au! BufEnter * call <SID>MakeBuff()
+let matches  = 0
+while matches <= 2
+  let matches += 1
+  au! CursorHold * call <SID>LineMatch{matches}()
```
Comment (Petesta) [inline plugin/poi.vim]: Interpolation here to call the correct `LineMatch#` function?

File: plugin/poi.vim [hunk 3]
```diff
@@ -1,32 +1,56 @@
...
+let g:poi_bg1 = 88
+let g:poi_fg1 = 15
+let g:g_poi_bg1 = "#870000"
+let g:g_poi_fg1 = "#ffffff"
+" yellow/purple
+let g:poi_bg2 = 226
+let g:poi_fg2 = 93
+let g:g_poi_bg2 = "#ffff00"
+let g:g_poi_fg2 = "#8700ff"
+" organe/lightblue
```
Comment (Petesta) [inline plugin/poi.vim]: organe => orange

**Comment by Petesta** [comment]
Overall I think this is super sweet! We should definintely dry up the `LineMatch#{NUMBER}` function but for now this is awesome! We can come back once we've figured out a way to do so.

**Pull request merged**